Panic on instagram failure

Wrong ID in short code etc. Should be handled more grafefully.

```
ERROR 2017/02/17 15:10:23 Retry #1 for https://api.instagram.com/oembed/?url=https://instagram.com/p/BN1pgzujlyP-/&hidecaption=0 and sleeping for 2s
panic: reflect: Zero(nil) [recovered]
	panic: reflect: Zero(nil)

goroutine 356 [running]:
panic(0x636960, 0xc422ff9460)
	/usr/local/Cellar/go/1.7.4/libexec/src/runtime/panic.go:500 +0x1a1
text/template.errRecover(0xc423e71830)
	/usr/local/Cellar/go/1.7.4/libexec/src/text/template/exec.go:146 +0x18e
panic(0x636960, 0xc422ff9460)
	/usr/local/Cellar/go/1.7.4/libexec/src/runtime/panic.go:458 +0x243
reflect.Zero(0x0, 0x0, 0x656440, 0xc42208f1e8, 0x94)
	/usr/local/Cellar/go/1.7.4/libexec/src/reflect/value.go:2104 +0x13d
text/template.(*state).validateType(0xc423e717b0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.7.4/libexec/src/text/template/exec.go:681 +0x5c1
text/template.(*state).evalArg(0xc423e717b0, 0x6914c0, 0xc4236c7200, 0x16, 0x0, 0x0, 0xa38d80, 0xc42020e5a0, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.7.4/libexec/src/text/template/exec.go:727 +0x72b
text/template.(*state).evalChainNode(0xc423e717b0, 0x6914c0, 0xc4236c7200, 0x16, 0xc4211d8580, 0xc4201fd7b0, 0x1, 0x1, 0x0, 0x0, ...)
```

--- a/tpl/tplimpl/template_embedded.go
+++ b/tpl/tplimpl/template_embedded.go
@@ -60,7 +60,7 @@ func (t *GoHTMLTemplate) EmbedShortcodes() {
 {{ end }}`)
 	t.AddInternalShortcode("gist.html", `<script src="//gist.github.com/{{ index .Params 0 }}/{{ index .Params 1 }}.js{{if len .Params | eq 3 }}?file={{ index .Params 2 }}{{end}}"></script>`)
 	t.AddInternalShortcode("tweet.html", `{{ (getJSON "https://api.twitter.com/1/statuses/oembed.json?id=" (index .Params 0)).html | safeHTML }}`)
-	t.AddInternalShortcode("instagram.html", `{{ if len .Params | eq 2 }}{{ if eq (.Get 1) "hidecaption" }}{{ (getJSON "https://api.instagram.com/oembed/?url=https://instagram.com/p/" (index .Params 0) "/&hidecaption=1").html | safeHTML }}{{ end }}{{ else }}{{ (getJSON "https://api.instagram.com/oembed/?url=https://instagram.com/p/" (index .Params 0) "/&hidecaption=0").html | safeHTML }}{{ end }}`)
+	t.AddInternalShortcode("instagram.html", `{{ if len .Params | eq 2 }}{{ if eq (.Get 1) "hidecaption" }}{{ with getJSON "https://api.instagram.com/oembed/?url=https://instagram.com/p/" (index .Params 0) "/&hidecaption=1" }}{{ .html | safeHTML }}{{ end }}{{ end }}{{ else }}{{ with getJSON "https://api.instagram.com/oembed/?url=https://instagram.com/p/" (index .Params 0) "/&hidecaption=0" }}{{ .html | safeHTML }}{{ end }}{{ end }}`)
 }
 
 func (t *GoHTMLTemplate) EmbedTemplates() {

